pubtools/_pulp/tasks/copy_repo.py

```diff
@@ -0,0 +1,324 @@
+        # It's crucial that the original order is maintained.
+        found_repos = []
+        src_repo = None
+        dest_repo = None
+
+        # Eagerly load all repos to fail early if the user passed any nonexistent repo.
+        search = self.pulp_client.search_repository(Criteria.with_id(repos))
+        for repo in search.result():
+            found_repos.append(repo)
+            if repo.id == repos[0]:
+                src_repo = repo
+            if repo.id == repos[1]:
+                dest_repo = repo
+
+        # Bail out if user requested repos which don't exist
+        missing = set(repos) - set([repo.id for repo in found_repos])
+        missing = sorted(list(missing))
+        if missing:
+            self.fail("Requested repo(s) don't exist: %s", ", ".join(missing))
+
+        # Bail out if we'd be processing any container image repos.
+        # We don't support this now because:
+        #
+        # - recording push items isn't implemented yet and it's not clear
+        #   how to implement it (as we traditionally used docker-image-*.tar.gz
+        #   filenames from brew as push item filename, but those aren't available
+        #   in pulp metadata)
+        #
+        # - no known use-case for clearing them
+        #
+        container_repo_ids = sorted(
+            [
+                repo.id
+                for repo in found_repos
+                if isinstance(repo, ContainerImageRepository)
+            ]
+        )
+        if container_repo_ids:
+            self.fail(
+                "Container image repo(s) provided, not supported: %s"
+                % ", ".join(sorted(container_repo_ids))
+            )
+
+        return src_repo, dest_repo
+
+    def log_copy(self, copied_repo):
+        # Given a repo which has been copied, log some messages
+        # summarizing the copied unit(s)
+        content_types = {}
+
+        for task in copied_repo.tasks:
+            for unit in task.units:
+                type_id = unit.content_type_id
+                content_types[type_id] = content_types.get(type_id, 0) + 1
+
+        task_ids = ", ".join(sorted([t.id for t in copied_repo.tasks]))
+        repo_id = copied_repo.repo.id
+        if not content_types:
+            LOG.warning("%s: no content copied, tasks: %s", repo_id, task_ids)
+        else:
+            copied_types = []
+            for key in sorted(content_types.keys()):
+                copied_types.append("%s %s(s)" % (content_types[key], key))
+            copied_types = ", ".join(copied_types)
+
+            LOG.info("%s: removed %s, tasks: %s", repo_id, copied_types, task_ids)
+
+        return copied_repo
+
+    @step("Copy content")
+    def copy_content(self, src_repo, dest_repo):
+        futures = []
+        content_types = self.content_type or ["None"]
+        for t in content_types:
+            crit = None
+            if t == "iso":
```

Comment:
It feels like "erratum" might be missing here.  Generally if we're copying RPMs, we'd want to be copying the erratum they belong to as well, so it seems quite suspicious to find code which supports ("rpm", "srpm") but not "erratum".

Reply:
Thanks!